### PR TITLE
Use checkout on 5.2, normalise http-examples [7xx]

### PR DESCRIPTION
**Pull request by mauritsvanrees**
This is a backport of my PR #1215 to branch 7.x.x.
It also updates 5.2.x to use Plone 5.2.5, including using the newer `plone.dexterity`. We normalise its unpredictable behavior schema names in the http examples.

This PR is best viewed with `?w=1` at the end of the url, to ignore whitespace changes, especially with the commit that updates the `http-examples`, which is actually a whitespace-only change.

### CODE REVIEW

**Review by tisto — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
@jenkins-plone-org please run jobs

**Review by avoinea — APPROVED** [review]

File: buildout.cfg
Comment (wesleybl) [inline buildout.cfg:9]: I found that issue #1018 occurs because of `plone.dexterity 2.9.10+`. More specifically after https://github.com/plone/plone.dexterity/pull/137

Strangely, it only occurs in Python 2.

The error occurred in Jenkins, but it did not occur in `plone.restapi` GA, because the `plone.restapi` buildout was pinned to `plone.dexterity 2.9.8`.

After the merge of this PR, the error can happen in GA as well.

**Review by wesleybl — COMMENTED** [review]
I left some comments.

File: plone-5.2.x.cfg
Comment (wesleybl) [inline plone-5.2.x.cfg:29]: I found that issue #1018 occurs because of plone.dexterity 2.9.10+. More specifically after plone/plone.dexterity#137

Strangely, it only occurs in Python 2.

The error occurred in Jenkins, but it did not occur in plone.restapi GA, because the plone.restapi buildout was pinned to plone.dexterity 2.9.8.

After the merge of this PR, the error can happen in GA as well.

File: src/plone/restapi/tests/test_documentation.py
Comment (mauritsvanrees) [reply src/plone/restapi/tests/test_documentation.py:100]: I have now removed the hardcoded ZSERVER_PORT in the PR.

A good thing is, that this makes it possible to introduce tox and run it in parallel, if we want. (That is outside the scope of this PR though.)

File: src/plone/restapi/tests/test_documentation.py
Comment (mauritsvanrees) [reply src/plone/restapi/tests/test_documentation.py]: I have updated the PR to use a name closer to what newer dexterity gives us.

File: plone-5.2.x.cfg
Comment (mauritsvanrees) [reply plone-5.2.x.cfg:29]: I guess with newer dexterity more test failures could indeed happen on GH actions. If so, this is good: it should bring us closer to fixing the underlying problem, whether that is in 'real' code or test code.

File: src/plone/restapi/tests/test_documentation.py
Comment (wesleybl) [inline src/plone/restapi/tests/test_documentation.py:546]: I think we can be less invasive here. That is, replacing only the really dynamic part. I suggest:

```suggestion
        document_schema_re = re.compile(
            r"^plone.dexterity.schema.generated.plone_5_\d*_2_\d*_0_Document$"
        )
```
So, if we have changes like what happened in `plone.dexterity 2.10+`, the tests will fail.

Comment (mauritsvanrees) [reply src/plone/restapi/tests/test_documentation.py:546]: Why would we do this?
I have no idea if the 5, 2 and 0 are always here or if they can differ as well. Seems pretty random to me.
This is an internal dexterity detail. If it changes, I do *not* want these tests to fail.

Comment (wesleybl) [reply src/plone/restapi/tests/test_documentation.py:546]: They will only change if we have structural changes like the ones that occurred not `plone.dexterity 2.10`. They don't change "via parameter". In this case, I find it valid for the test to fail, as failed. If we had this regular expression, before the change from `plone.dexterity 2.10`, the tests would not fail, even with the name pattern changing. And I think that in this case it should fail.

`schemaName` after this line:

https://github.com/plone/plone.dexterity/blob/35a12ab726b76ee5527ee467974917bc1fbdca0c/plone/dexterity/fti.py#L262

it's always something like:

`plone_5_1631640310_2_6213937_0_Document`

Comment (mauritsvanrees) [reply src/plone/restapi/tests/test_documentation.py:546]: I see the 5 and 0 come from the [`SchemaNameEncoder`](https://github.com/plone/plone.dexterity/blob/35a12ab726b76ee5527ee467974917bc1fbdca0c/plone/dexterity/schema.py#L295-L321).
I see no value in having a test in `plone.restapi` fail when this class changes.

Comment (wesleybl) [reply src/plone/restapi/tests/test_documentation.py:546]: We can have more similar values to the real ones in the documentation. I think that's why we have testing to generate documentation, rather than just writing the documentation.

Comment (mauritsvanrees) [reply src/plone/restapi/tests/test_documentation.py]: Fine, since no one stepped up to argue in my favour, I have changed it: commit 8921becee228c0fa2c43f31ed5d4731119a53bc7.

I think it is fine to merge now, right? Well, once everything is green.

**Comment by mauritsvanrees** [comment]
@jenkins-plone-org please run jobs

**Review by wesleybl — APPROVED** [review]
It would be nice if the changes only made on this branch are also made on the master.

**Comment by mauritsvanrees** [comment]
> It would be nice if the changes only made on this branch are also made on the master.

Will do.

**Comment by mauritsvanrees** [comment]
Done: https://github.com/plone/plone.restapi/pull/1226

**Pull request merged**
